mantidproject/mantid
# Reflectometry: Multiple Row Checks

**Description of work.**
    Handle multiple row matches in exp. settings
    
    This adds handling for if we match multiple theta AND title values.
    Since we don't want to write our own checking logic in the view, we do
    this once a user hits process so that we can check the regex values.
    
    This is typically seen if a user has a greedy regex (e.g. ".*") and a
    more specific matcher than conflict.

**To test:**
- Enter a group name `A`, enter an angle of `0.1` and the run as `13460`
- Copy paste the group, rename the other groups to `AA` and `AAA`
- In experiment settings tab add another row. In the first set the title to `A.*` and the angle to 0.1, in the second set the title to `AA.*` and angle 0.1
- Try processing in the main tab this should give a sensible warning
- In the experiment settings tab, try changing the title matchers to `AA.*` and `AAA.*`
- This should process the first run (using the defaults) and give a warning on the second

Fixes #33516 

*This does not require release notes* because this is part of ongoing, unreleased work in #31795 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
